website/addons/gdrive/static/gdriveUserConfig.js

```diff
@@ -0,0 +1,126 @@
+
+/**
+* View model that controls the Goolge Drive configuration on the user settings page.
+*/
+'use strict';
+var ko = require('knockout');
+require('knockout-punches');
+ko.punches.enableAll();
+var $ = require('jquery');
+var Raven = require('raven-js');
+var bootbox = require('bootbox');
+
+var language = require('osfLanguage').Addons.gdrive;
+var osfHelpers = require('osfHelpers');
+
+    var ViewModel = function(url) {
+        var self = this;
+        self.userHasAuth = ko.observable(false);
+        self.loaded = ko.observable(false);
+        self.urls = ko.observable();
+        self.username = ko.observable();
+
+        //Helper-class variables
+        self.message = ko.observable('')
+        self.messageClass = ko.observable('text-info');
+
+
+        $.ajax({
+            url : url, type: 'GET', dataType: 'json',
+            success: function(response) {
```

Comment:
Use the Promise interface. 

http://cosdev.readthedocs.org/en/latest/style_guides/javascript.html#ajax